src/content/url-handling.js

```diff
@@ -81,10 +82,24 @@ export function urlFromState(urlState: URLState) {
   return pathname + (qString ? '?' + qString : '');
 }
 
-export function stateFromCurrentLocation(): URLState {
-  const pathname = window.location.pathname;
-  const qString = window.location.search.substr(1);
-  const hash = window.location.hash;
```

Comment:
I made this function take a parameter to make it easier to use in tests.

And looks like it brought some more flow coverage -- maybe window.location was actually `any` ?